Bucketlist checkpoint 3

wmahad opened the pull request:
Implementation of the bucketlist API in django and Frontend Interface with angular

wmahad committed:
(chore) Add .gitignore

ignores certain file types
i.e. *.pyc .env

wmahad committed:
(chore) Add Testing Configs

add .travis.yml file
add .coverage file

wmahad committed:
(chore) Add Tests

add tests for models
add test_models.py

wmahad committed:
(chore) Modify .gitignore

add migrations in .gitignore

wmahad committed:
(feature) Add API Files

add models.py
add urls.py
add views.py
add serializers.py

wmahad committed:
(feature) Add UI

- add login and signup user interfaces
- add bucketlist and item UI
- add permissions for serialisers

wmahad committed:
(chore) Add Comments

- add class doc-strings and coments
- modify urls to restrict api access
  in a browser

wmahad committed:
(fix) Indentention errors

-remove indentation errors from .py files
-Add logic for logged in user in angular

wmahad committed:
(fix) remove python 3,2~ compatibility

wmahad committed:
(fix) Migration Issue

wmahad committed:
(fix) Add auth, bucketlist Migrations

wmahad committed:
(fix) Add auth migration

wmahad committed:
(chore) Remove Migrations from .gitignore

wmahad committed:
(chore) Add Documentation

-Add Swagger Integration
-Add url config for Swagger
-Add 'rest_framework_swagger' to settings.py
-Add Procfile for heroku

wmahad committed:
Add README.md

wmahad committed:
(fix) Deployment to heroku

andela-anandaa commented on README.md:
```diff
@@ -0,0 +1,82 @@
+ Ensure your create `DATABASE_URL` and `SECRET` in your env variables.
+
+5. ######Managing Database set up
+
```
Should not skip a line, for the numbering

wmahad committed:
(fix) Update README.md

wmahad committed:
(chorre) Add API-endpoint tests

- Add tests for the api
- modify bucketlist view
- modify items view
- Add toasts to app

wmahad committed:
(fix) UI/UX flows

-Add favicon
-Edit Item View
-Edit Bucketlist View

wmahad committed:
removed ipdb from requirements.txt
